examples/contrib/mnist/mnist_with_wandb_logger.py

```diff
@@ -0,0 +1,191 @@
+from ignite.handlers import ModelCheckpoint
+
+from ignite.contrib.handlers.wandb_logger import *
+
+
+class Net(nn.Module):
+    def __init__(self):
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
+        self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
+        self.conv2_drop = nn.Dropout2d()
+        self.fc1 = nn.Linear(320, 50)
+        self.fc2 = nn.Linear(50, 10)
+
+    def forward(self, x):
+        x = F.relu(F.max_pool2d(self.conv1(x), 2))
+        x = F.relu(F.max_pool2d(self.conv2_drop(self.conv2(x)), 2))
+        x = x.view(-1, 320)
+        x = F.relu(self.fc1(x))
+        x = F.dropout(x, training=self.training)
+        x = self.fc2(x)
+        return F.log_softmax(x, dim=-1)
+
+
+def get_data_loaders(train_batch_size, val_batch_size):
+    data_transform = Compose([ToTensor(), Normalize((0.1307,), (0.3081,))])
+
+    train_loader = DataLoader(
+        MNIST(download=True, root=".", transform=data_transform, train=True), batch_size=train_batch_size, shuffle=True
+    )
+
+    val_loader = DataLoader(
+        MNIST(download=False, root=".", transform=data_transform, train=False), batch_size=val_batch_size, shuffle=False
+    )
+    return train_loader, val_loader
+
+
+def run(train_batch_size, val_batch_size, epochs, lr, momentum):
+    train_loader, val_loader = get_data_loaders(train_batch_size, val_batch_size)
+    model = Net()
+    device = "cpu"
+
+    if torch.cuda.is_available():
+        device = "cuda"
+
+    optimizer = SGD(model.parameters(), lr=lr, momentum=momentum)
+    criterion = nn.CrossEntropyLoss()
+    trainer = create_supervised_trainer(model, optimizer, criterion, device=device)
+
+    if sys.version_info > (3,):
+        from ignite.contrib.metrics.gpu_info import GpuInfo
+
+        try:
+            GpuInfo().attach(trainer)
+        except RuntimeError:
+            print(
+                "INFO: By default, in this example it is possible to log GPU information (used memory, utilization). "
+                "As there is no pynvml python package installed, GPU information won't be logged. Otherwise, please "
+                "install it : `pip install pynvml`"
+            )
+
+    metrics = {"accuracy": Accuracy(), "loss": Loss(criterion)}
+
+    train_evaluator = create_supervised_evaluator(model, metrics=metrics, device=device)
+    validation_evaluator = create_supervised_evaluator(model, metrics=metrics, device=device)
+
+    @trainer.on(Events.EPOCH_COMPLETED)
+    def compute_metrics(engine):
+        train_evaluator.run(train_loader)
+        validation_evaluator.run(val_loader)
+
+    wandb_logger = WandBLogger(
+        project="pytorch-ignite-integration",
+        name="ignite-mnist-example",
+        config={
+            "train_batch_size": train_batch_size,
+            "val_batch_size": val_batch_size,
+            "epochs": epochs,
+            "lr": lr,
+            "momentum": momentum,
+        },
+    )
+
+    def iteration(engine):
```

Comment:
We can use `global_step_from_engine` instead of `iteration`:
`global_step_transform=global_step_from_engine(trainer)`

Reply:
The problem is that `global_step_from_engine` will use the **iteration** number for handlers called on `Events.ITERATION_COMPLETED`, but **epoch** number for handlers called on `Events.EPOCH_COMPLETED`. W&B does not allow to log events with a smaller step than previously logged, so I need to make sure to always use the iteration number.

Reply:
OK, I see what you mean. So, we can not log at all any epoch number if we log iterations ?

Reply:
Exactly, at least not directly. The workaround (as described in the W&B docs) is to log epochs as "metric", and then select it as X-axis in the W&B web interface. Not sure how to facilitate this in the Logger here.

Thanks for the simpler `global_step_transform`, will add it!